compatibility.js

```diff
@@ -0,0 +1,65 @@
+function versionToInt (version) {
+    // convert v2 to 2
+    return parseInt(version.replace('v', ''))
+}
+
+function removeEolFeatures (config, prevConfig, version) {
```

Comment:
This function is very hard to follow. At least document it more but I think it should be rethought to prevent issues.

My understanding is you're passing in `config` which is the current live config and removing features based on the current compat function.

- These side effects of editing two object references solve the problem but it feels unrelated to the current config you're operating on.

I would suggest it would be much simpler to follow to have a separate step ran after the `compatFunctions` to remove the relevant attributes/features from that config version only.

- That means that each call site `compatFunction` wouldn't have to wire in the call to this method potentially passing the wrong arguments.
- I would place this call just before the call to `addHashToFeatures(compatConfig)`
- Instead after running `compatFunction` `v2` you would delete anything that was EOL'd in v1
- This step also needs to apply to the current version so it doesn't really belong in the compat function.

Reply:
I toyed around with this approach and it's probably more correct. I ended up here after trying to generalize the v1 function. I'll rework it this way though as it does seem more clean.

Reply:
I'll be around tomorrow to help/explain further in case what I wrote makes no sense as it's late here haha 💤